src/openrct2/core/FileScanner.cpp

```diff
@@ -277,15 +277,15 @@ class FileScannerUnix final : public FileScannerBase
     void GetDirectoryChildren(std::vector<DirectoryChild>& children, std::string_view path) override
     {
         struct dirent** namelist;
-        int32_t count = scandir(path.c_str(), &namelist, FilterFunc, alphasort);
+        int32_t count = scandir(path.data(), &namelist, FilterFunc, alphasort);
```

Comment:
Is this guaranteed to be null-terminated?

Reply:
Short answer: It _is_ possible that the actual data isn't null-terminated, but you'd have to do your best to get there.

The `string_view::data` member function itself doesn't guarantee that its data is null-terminated, however `path` gets implicitly constructed from null-terminated strings, which under the hood uses `strlen` to determine the length. You'd have to manually provide a length, take a substring of a path, or do other shenanigans to make the length it knows different from the actual string.

Reply:
This also has the added bug of letting you read beyond the size of the path.
```
std::string thing("pathtoitem/ignorethisbit/")
std::string_view thingView = thing;
std::string_view subThingView = thingView.substr(0, 11); // Get just the "pathtoitem/"
GetDirectoryChildren(output, subThingView); // Error scans "pathtoitem/ignorethisbit/"
```

Reply:
Yeah, that's what I was saying, an unlikely scenario. But I agree that it's better to avoid this at all.